src/SFML/Window/Unix/KeyboardImpl.cpp

```diff
@@ -0,0 +1,586 @@
+        case XK_u:                return sf::Keyboard::ScanU;
+        case XK_v:                return sf::Keyboard::ScanV;
+        case XK_w:                return sf::Keyboard::ScanW;
+        case XK_x:                return sf::Keyboard::ScanX;
+        case XK_y:                return sf::Keyboard::ScanY;
+        case XK_z:                return sf::Keyboard::ScanZ;
+        case XK_1:                return sf::Keyboard::ScanNum1;
+        case XK_2:                return sf::Keyboard::ScanNum2;
+        case XK_3:                return sf::Keyboard::ScanNum3;
+        case XK_4:                return sf::Keyboard::ScanNum4;
+        case XK_5:                return sf::Keyboard::ScanNum5;
+        case XK_6:                return sf::Keyboard::ScanNum6;
+        case XK_7:                return sf::Keyboard::ScanNum7;
+        case XK_8:                return sf::Keyboard::ScanNum8;
+        case XK_9:                return sf::Keyboard::ScanNum9;
+        case XK_0:                return sf::Keyboard::ScanNum0;
+        case XK_space:            return sf::Keyboard::ScanSpace;
+        case XK_minus:            return sf::Keyboard::ScanHyphen;
+        case XK_equal:            return sf::Keyboard::ScanEquals;
+        case XK_bracketleft:      return sf::Keyboard::ScanLBracket;
+        case XK_bracketright:     return sf::Keyboard::ScanRBracket;
+        case XK_backslash:        return sf::Keyboard::ScanBackslash;
+        case XK_semicolon:        return sf::Keyboard::ScanSemicolon;
+        case XK_apostrophe:       return sf::Keyboard::ScanQuote;
+        case XK_grave:            return sf::Keyboard::ScanGraveAccent;
+        case XK_comma:            return sf::Keyboard::ScanComma;
+        case XK_period:           return sf::Keyboard::ScanPeriod;
+        case XK_slash:            return sf::Keyboard::ScanForwardSlash;
+        case XK_less:             return sf::Keyboard::ScanReverseSolidus;
+        default:                  return sf::Keyboard::ScanUnknown;
+    }
+}
+
+
+////////////////////////////////////////////////////////////
+void initMapping()
+{
+    for (int i = 0; i < 256; ++i)
+        scancodeToKeycode[i] = NullKeyCode;
+
+    for (int i = 0; i < sf::Keyboard::ScanCodeCount; ++i)
+        keycodeToScancode[i] = sf::Keyboard::ScanUnknown;
+
+    Display* display = sf::priv::OpenDisplay();
+
+    // Find the X11 key code -> SFML key code mapping
+    // This code was inspired by GLFW implementation
+
+    char name[XkbKeyNameLength + 1];
+    XkbDescPtr desc = XkbGetMap(display, 0, XkbUseCoreKbd);
+    XkbGetNames(display, XkbKeyNamesMask, desc);
+
+    sf::Keyboard::Scancode sc;
+
+    for (int keycode = desc->min_key_code; keycode <= desc->max_key_code; ++keycode)
+    {
+        std::memcpy(name, desc->names->keys[keycode].name, XkbKeyNameLength);
+        name[XkbKeyNameLength] = '\0';
+
+        if (strcmp(name, "TLDE") == 0)      sc = sf::Keyboard::ScanGraveAccent;
```

Comment:
I would align the `(strcmp(name, "TLDE") == 0)` with the lines below just for that extra touch.